tests/test_audio_processor.cpp

```
@@ -110,3 +111,38 @@ TEST(AudioProcessor, StereoToMonoAndResample)
+	std::vector<short> data = LoadAudioFile("data/test_stereo_44100.raw");
+
+	AudioBuffer expected;
+	AudioProcessor reference(11025, &expected);
+	reference.Reset(44100, 2);
+	reference.Consume(data.data(), data.size());
+	reference.Flush();
+
+	// Sizes that are not multiples of the channel count, so that every call
+	// but the last leaves a partial frame behind. What the caller passes per
+	// call must not change the result.
+	const size_t chunk_sizes[] = { 1, 2, 3, 5, 7, 17, 511, 997, 4096 };
```

Clang (14.0.6)

[warning] do not declare C-style arrays, use std::array<> instead

(modernize-avoid-c-arrays)